### ISSUE TITLE
Improve performance of get_entity_by_id

### ISSUE THREAD

**Issue by yhmo** [milvus-io/milvus#2426]
**Describe the bug**
get_entity_by_id performance is not good with data increase and partition increase.

Avoid unnecessary file reading from disk and reduce SQL execution times to improve it performance

**Comment by yhmo** [comment]
dim=512, 1000 partitions, insert 10000 vectors to each partition.
call get_entity_by_id to find 1 vector, time cost:1.4 second

**Comment by yhmo** [comment]
fixed by pr2438

**Comment by yhmo** [comment]
After the fix, get_entity_by_id time cost is about 0.3 second

**Comment by del-zhenwu** [comment]
Redash query params:
```
{
    "collection": "doc",
    "query": {
        "metrics.type": "get_ids_performance",
        "server.value.mode": "CPU",
        "collection.dataset_name": "sift_10m_1024_128_l2",
        "run_params.ids_num": 100
    },
    "fields": {
        "datetime": 1,
        "metrics.value.total_time": 2,
        "metrics.value.avg_time": 3
    },
    "sort": [{
        "name": "run_id",
        "direction": -1
    }],
    "limit": 100
}
```
